### PR TITLE
Asset Info Plugin

### PR DESCRIPTION
**Pull request by toastercup**
- Implement storage of thumbnail/style URLs
- Asset wizard info box
- Asset thumbnail in index

### CODE REVIEW

**Comment by ElliottAYoung** [comment]
Are we merging this into core?
I thought we were making it a separate plugin?

**Comment by toastercup** [comment]
@ElliottAYoung The PR's name is a bit misleading: this is supporting infrastructure so Asset Info Plugins can work.

File: app/models/asset_field_type.rb [hunk 1]
```diff
@@ -115,11 +117,19 @@ def validate_asset_content_type
...
+    end
+  end
+
   def existing_metadata
     metadata.except!(:existing_data)
 
-    unless @existing_data.empty?
-      metadata[:path] = updated_url(@existing_data['asset']['url'])
+    unless existing_data.empty?
+      metadata[:path] = updated_url(existing_data['asset']['url'])
```
Comment (ElliottAYoung) [inline app/models/asset_field_type.rb]: This is gonna merge conflict, but it's inevitable! 😆😆😆😆😆😆😆😆😆😆😆😆😆😆😆

**Review by ElliottAYoung — APPROVED** [review]

**Comment by toastercup** [comment]
rebased; merging.

**Pull request merged**